rfd/0147-repo-fork-syncing.md

```diff
@@ -0,0 +1,60 @@
+---
+authors: Mike Jensen (mike.jensen@goteleport.com)
+state: draft
+---
+ 
+# RFD 144 - Automatic Repo Syncing
+ 
+## What
+ 
+This RFD describes how we can sync a repo continuously and in an automatic way.  This can be used to provide private copies of a repo.
+ 
+## Why
+ 
+We currently face a few frictions which could be addressed through a more robust repo sync mechanism:
+1. Our internal `teleport-private` fork used for security development must be manually updated
+2. Some tooling (Dependabot) does not allow scanning for non-default branches, necessitating manual efforts to manage our release branches
+3. Because much of the CI and scanning configuration is committed into the repo, we face noise from duplicate alerts when we use a repo for testing or as an internal fork
+ 
+## Details
+ 
+As such we need a solution that can provide the following:
+* Automatic and regular updates to a repo to ensure it's in sync
+* The ability to apply changes to the upstream repo (for example disable Actions that we don't want running on the repo copy, or enable additional scanning / actions)
+ 
+### Branch Structure
+ 
+* `main` - This will be the repos default branch.  This branch will contain the upstream history in addition to commits needed for the sync process or other custom changes on the repo copy
```

Comment:
```suggestion
* `main` - This will be the repo's default branch.  This branch will contain the upstream history in addition to commits needed for the sync process or other custom changes on the repo copy
```

How exactly will you keep `main` in a fork in sync with `master` in the upstream?